src/lib/lss.ts

```diff
@@ -86,15 +86,38 @@ export async function createSplitsXml(config: Config): Promise<string> {
     } = config;
 
     const splitDefinitions = parseSplitsDefinitions();
-    const icons = await getIconData();
+    const iconLocations = await getIconLocations();
+    const iconFiles = new Set<string>();
+    const iconData = new Map<string, string>();
+
+    splitIds.forEach(splitId => {
+        const file = iconLocations.get(splitId)?.file;
+        if (file) {
+            iconFiles.add(file);
+        }
+    });
+    await Promise.all(
```

Comment:
we don't need .then in async/await land
```suggestion
    const iconLookups = await Promise.all(
```

Reply:
Looks so clean all of a sudden :)